common-arcana.lic

```diff
@@ -46,23 +46,42 @@ module DRCA
 
   def start_khris(khris, settings)
     khris
-      .map { |name| "Khri #{name.capitalize}" }
-      .each do |name|
-        activate_khri?(settings.kneel_khri, name)
+      .each do |khri_set|
+        activate_khri?(settings.kneel_khri, khri_set)
       end
   end
 
+  # Needs to handle (based on current usage):
+  # - Hasten
+  # - Delay Hasten
+  # - Hasten Focus
+  # - Delay Hasten Focus
+  # - Khri Hasten
+  # - Khri Delay Hasten
+  # - Khri Delay Hasten Focus
   def activate_khri?(settings_kneel, ability)
-    return false if DRSpells.active_spells[ability]
+    abilities = ability.split.map(&:capitalize)
 
-    kneel = kneel_for_khri?(settings_kneel, ability)
+    # Standardize for with/without 'Khri' on the front
+    abilities = abilities.drop(1) if abilities.first.casecmp('khri') == 0
+
+    # Handling for 'Delay'
+    should_delay = abilities.first.casecmp('delay') == 0
+    abilities = abilities.drop(1) if should_delay
+
+    # Check each khri in the list against Active Spells, Drop any that are active
+    needed_abilities = abilities.select { |ability_to_check| DRSpells.active_spells["Khri #{ability_to_check}"].nil? }
+    return true if needed_abilities.empty?
+
+    kneel = needed_abilities.any? { |ability_to_check| kneel_for_khri?(settings_kneel, ability_to_check) }
     DRC.retreat if kneel
     DRC.bput('kneel', 'You kneel', 'You are already', 'You rise') if kneel && !kneeling?
-    result = DRC.bput(ability, get_data('spells').khri_preps)
+
+    result = DRC.bput("Khri #{should_delay ? 'Delay ' : ''}#{needed_abilities.join(' ')}", get_data('spells').khri_preps)
     waitrt?
     DRC.fix_standing
 
-    ['Your body is willing', 'You have not recovered'].include?(result)
+    return !['Your mind and body are willing', 'Your body is willing', 'You have not recovered'].include?(result)
```

Comment:
6 of one, half dozen of another. What are your thoughts on changing `include?` to `none?` ?
```suggestion
    return ['Your mind and body are willing', 'Your body is willing', 'You have not recovered'].none?(result)
```

Reply:
I don't see that on this version of ruby:
https://ruby-doc.org/core-2.5.5/Array.html

Am I just missing it?

Reply:
It’s inherited from the Enumerable parent class: https://ruby-doc.org/core-2.5.5/Enumerable.html#method-i-none-3F

I missed it originally too; those docs don’t make it clear what methods are available due to inheritance, you have to go to the parent classes to find them :(

Reply:
Well, that's decidedly annoying. It shows some methods, but not all of them - made me assume all that worked were documented. I'll poke around at it.